theano/gpuarray/c_code/dnn_sptf_gi.c

```diff
@@ -0,0 +1,172 @@
+#section support_code_struct
+
+cudnnTensorDescriptor_t APPLY_SPECIFIC(xdesc);
+cudnnTensorDescriptor_t APPLY_SPECIFIC(dxdesc);
+cudnnTensorDescriptor_t APPLY_SPECIFIC(dydesc);
+
+#section init_code_struct
+
+APPLY_SPECIFIC(xdesc) = NULL;
+APPLY_SPECIFIC(dxdesc) = NULL;
+APPLY_SPECIFIC(dydesc) = NULL;
+
+{
+    cudnnStatus_t err = CUDNN_STATUS_SUCCESS;
+    err = cudnnCreateTensorDescriptor( &APPLY_SPECIFIC(xdesc) );
+    if ( err != CUDNN_STATUS_SUCCESS )
+    {
+        PyErr_Format( PyExc_MemoryError,
+            "GpuDnnTransformerGradI: failed to allocate cuDNN tensor descriptor xdesc: %s",
+            cudnnGetErrorString( err ) );
+        FAIL;
+    }
+
+    err = cudnnCreateTensorDescriptor( &APPLY_SPECIFIC(dxdesc) );
+    if ( err != CUDNN_STATUS_SUCCESS )
+    {
+        PyErr_Format( PyExc_MemoryError,
+            "GpuDnnTransformerGradI: failed to allocate cuDNN tensor descriptor dxdesc: %s",
+            cudnnGetErrorString( err ) );
+        FAIL;
+    }
+
+    err = cudnnCreateTensorDescriptor( &APPLY_SPECIFIC(dydesc) );
+    if ( err != CUDNN_STATUS_SUCCESS )
+    {
+        PyErr_Format( PyExc_MemoryError,
+            "GpuDnnTransformerGradI: failed to allocate cuDNN tensor descriptor dydesc: %s",
+            cudnnGetErrorString( err ) );
+        FAIL;
+    }
+}
+
+#section cleanup_code_struct
+
+if ( APPLY_SPECIFIC(xdesc) != NULL )
+    cudnnDestroyTensorDescriptor( APPLY_SPECIFIC(xdesc) );
+
+if ( APPLY_SPECIFIC(dxdesc) != NULL )
+    cudnnDestroyTensorDescriptor( APPLY_SPECIFIC(dxdesc) );
+
+if ( APPLY_SPECIFIC(dydesc) != NULL )
+    cudnnDestroyTensorDescriptor( APPLY_SPECIFIC(dydesc) );
+
+#section support_code_struct
+
+int
+APPLY_SPECIFIC(dnn_sptf_gi)(PyGpuArrayObject * input,
+                            PyGpuArrayObject * theta,
```

Comment:
It seems you don't need theta in this code. If so, it must be removed from here and from Op's inputs in Python code.